File: Trainning_LeetCode/104. Maximum Depth of Binary Tree.cpp
#include <iostream>
#include <vector>
#include <queue>
#include <stack>
#include <map>

using namespace std;

//Definition for a binary tree node.
struct TreeNode {
    int val;
    TreeNode *left;
    TreeNode *right;
    TreeNode() : val(0), left(nullptr), right(nullptr) {}
    TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
    TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
};

class Solution {
public:
    int maxDepth(TreeNode* root) {
        if(root == NULL)
            return 0;
        int depLeft = maxDepth(root->left);
        int depRight = maxDepth(root->right);
        return 1 + max(depLeft, depRight);
    }
};

int main()
{
    Solution s;
    s.climbStairs(2);
    return 0;
}

File: _Slides/Lession_11_12_Tree_01/SampleCode/_104_Maximum_Depth_of_Binary_Tree.java
public class _104_Maximum_Depth_of_Binary_Tree {
    public int maxDepth(TreeNode curNode) {
        if(curNode == null)
            return 0;

        int leftDepth = maxDepth(curNode.left) + 1;
        int rightDepth = maxDepth(curNode.right) + 1;
        return Math.max(leftDepth, rightDepth);
    }
}
